mreg_cli/api/models.py

```diff
@@ -521,7 +646,80 @@ def get_roles_with_atom(cls, name: str) -> list[Role]:
         :returns: A list of Role objects.
         """
         data = get_list(cls.endpoint(), params={"atoms__name__exact": name})
-        return [Role(**item) for item in data]
+        return [cls(**item) for item in data]
+
+    def add_atom(self, atom: str) -> bool:
+        """Add an atom to the role.
+
+        :param atom: The name of the atom to add.
+
+        :returns: True if the atom was added. False otherwise.
+        """
+        resp = post(Endpoint.HostPolicyRoleAddAtom.with_params(self.name), name=atom)
+        return resp.ok if resp else False
+
+    def remove_atom(self, atom: str) -> bool:
+        """Remove an atom from the role.
+
+        :param atom: The name of the atom to remove.
+
+        :returns: True if the atom was removed. False otherwise.
+        """
+        resp = delete(Endpoint.HostPolicyRoleRemoveAtom.with_params(self.name, atom))
+        return resp.ok if resp else False
+
+    def get_labels(self) -> list[Label]:
+        """Get the labels associated with the role.
+
+        :returns: A list of Label objects.
+        """
+        return [Label.get_by_id_or_raise(id_) for id_ in self.labels]
+
+    def add_label(self, label_name: str) -> Self:
+        """Add a label to the role.
+
+        :param label_name: The name of the label to add.
+
+        :returns: The updated Role object.
+        """
+        label = Label.get_by_name_or_raise(label_name)
+        if label.id in self.labels:
+            cli_warning(f"The role {self.name!r} already has the label {label_name!r}")
+
+        label_ids = self.labels.copy()
+        label_ids.append(label.id)
+        return self.patch({"labels": label_ids}, validate=False)
+
+    def remove_label(self, label_name: str) -> Self:
+        """Add a label to the role.
+
+        :param label_name: The name of the label to add.
+
+        :returns: The updated Role object.
+        """
+        label = Label.get_by_name_or_raise(label_name)
+        if label.id not in self.labels:
+            cli_warning(f"The role {self.name!r} doesn't have the label {label_name!r}")
+
+        label_ids = self.labels.copy()
+        label_ids.remove(label.id)
+        return self.patch({"labels": label_ids}, validate=False)
```

Comment:
`Role.{add,remove}_atoms` have different return types and implementations from `Role.{add,remove}_labels`. 

Should we try to _always_ have the full implementation in the models themselves, and then basically only call those methods from the command functions? So all info, warning and error messages are all printed from the model method itself?

Reply:
Keep `cli_info()` calls in commands or move those into models as well?

Reply:
Especially shoving everything into `Role.delete()` might lead to a bit of a clumsy implementation?

```python
    def delete(self) -> bool:
        """Delete the role."""
        if self.hosts:
            hosts = ", ".join(self.hosts)
            cli_error(f"Role {self.name!r} used on hosts: {hosts}")
        if super().delete():
            cli_info(f"Deleted role {self.name!r}", print_msg=True)
        else:
            cli_error(f"Failed to delete role {self.name!r}")
        return True # a bit contrived?
```

Reply:
I'd like to redo `cli_info` to either use `OutputManager.add_ok()` or something similar, and with that reimplement logging properly (if we want logging in the client). I was thinking of looking at this when I already started on https://github.com/unioslo/mreg-cli/pull/225

Reply:
Updated my comment just now btw

Reply:
> The rule for models could be that they never print, but may raise?

This still maps to being "cleaner and more correct with regards to separation of concern"? It's at least an easy rule to follow. :)